pyccel/codegen/printing/fcode.py

```diff
@@ -1076,6 +1076,21 @@ def _print_NumpyArray(self, expr):
             rhs_code = self._print(expr.arg)
         return rhs_code
 
+    def _print_NumpySign(self, expr):
+        str_x = self._print(expr.args[0])
+        kind  = self.print_kind(expr.args[0])
+
+        if isinstance(expr.dtype, NativeComplex):
+            ne_zero = f'(REALPART({str_x}) .ne. 0_{kind}) .or. (IMAGPART({str_x}) .ne. 0_{kind})'
+            lt_zero = f'((REALPART({str_x}) .eq. 0_{kind}) .and. IMAGPART({str_x}) .lt. 0_{kind}) .or. (REALPART({str_x}) .lt. 0_{kind})'
+        else:
+            ne_zero = f'({str_x} .ne. 0_{kind})'
+            lt_zero = f'({str_x} .lt. 0_{kind})'
+
+        merge_non_zero = f'merge(-1_{kind}, 1_{kind}, {lt_zero})'
+
+        return f'merge({merge_non_zero}, 0_{kind}, {ne_zero})'
+
```

Comment:
I agree with the principle, but I think that this is yet another example of something that ought to be handled by a dedicated routine.

At one point we planned to add a section of pyccel to contain (usually numpy) routines that need to be written in both Fortran and C. To avoid writing these manually for both languages and for any languages we may want to use in the future, we planned to write these routines in python and translate them using pyccel. These routines were supposed to be translated during the installation.

Unfortunately we still have not implemented this. Previously when looking at things which would benefit from this mechanism we have often said that we will leave them to one side for now to avoid doing the work twice.
One of the main things holding up the discussion about this mechanism was the lack of support for functions returning arrays which meant that most of the functions we would want to translate couldn't be translated. This is no longer an issue. I think it is time we revisited this. We would probably need to organise a meeting to discuss the implementation details.

If there is a chance of this being implemented soon then I don't know if it is worth duplicating the work if we already have a working implementation in Fortran

Reply:
I agree that we should have a meeting about this mechanism soon. But in this specific case I am not sure that pyccelizing a function written in Python would lead to C and Fortran code that is as efficient as the code written here by hand. It is worth trying, though!

Reply:
Ok, I don't know if this code should remain, but I managed to make a fortran module containing the sign functions for REAL, INTEGER and COMPLEX types, and it's working for arrays too.

Reply:
The new fortran module is located [here](https://github.com/pyccel/pyccel/blob/numpy.sign/pyccel/stdlib/numpy/numpy_f90.f90)

Reply:
Well done @aihya! I think we can use your new file for now. @EmilyBourne Is that fine with you?

Reply:
Fine with me

Reply:
Yes I assumed you were just asking if we could resolve this conversation